### ISSUE TITLE
Unexpected exception: 'utf-8' codec can't decode byte 0xc3 in position 0: unexpected end of data

### ISSUE THREAD

**Issue by fcastilloec** [althonos/ffpb#22]
I keep getting the following error: `Unexpected exception: 'utf-8' codec can't decode byte 0xc3 in position 0: unexpected end of data` no matter what I do. If I run it directly with `ffmpeg`, then there's no error whatsoever.

I installed v0.3.0 to check if the error was seen there but it's not. This is a problem with v0.4.0 only.
If there's a way I can debug the code to figure out what's happening.

PS: I installed `ffpb` through `pip3` on Ubuntu 20.04

**Comment by althonos** [comment]
Hi @fcastilloec ,

I suspect the issue comes from the [following lines](https://github.com/althonos/ffpb/blob/b0d39bd7491e88e10bd4bf81a78e82df13234597/ffpb.py#L160-L165) in `ffpb.py`. Everywhere else in the code `ffpb` uses the system encoding to decode things, except here where it assumes UTF-8. These were introduced in `v0.4.0`.

Could you please try replacing the following lines:
```python
while True:
    out = p.stderr.read(1).decode("utf-8")
    if out == '' and p.poll() != None:
        break
    if out != '':
        notifier(out)
```
with:
```python
while True:
    out = p.stderr.read(1)
    if out == b"" and p.poll() != None:
        break
    if out != b"":
        notifier(out)
```
and see if it works for you?

**Comment by fcastilloec** [comment]
That actually stopped the error message from appearing but now my terminal is frozen and I can see the `ffmpeg` process still running but with 0 CPU usage and _`N/A`_ for memory, it will stay there for hours (even though I'm not transcoding anything just copying/extracting a subtitle stream from an `mkv` file)
Running the same command with `ffmpeg` and `time`, I get: 
```
real	0m0.134s
user	0m0.082s
sys	0m0.051s
```

There might be some error happening when that `ffmpeg` process is so fast that `ffpb` doesn't have any info to work with, can that be an issue?

**Comment by fcastilloec** [comment]
@althonos Sorry! For some reason, I forgot to copy/paste the lines where `''` changes to `b""`, I only did the `decode` part.
After those changes. I don't see any errors and my console is not freezing anymore!
Sorry again for wasting some time... I should have read your code more carefully.

**Comment by althonos** [comment]
No problem, if it works i'll just release a patch! Cheers :smiley:

**Comment by althonos** [comment]
Fixed in `v0.4.1`.
